### ISSUE TITLE
TypeScript definition does not work when moduleResolution is node16 / nodenext

### ISSUE THREAD

**Issue by joshuaavalon** [hashicorp/next-mdx-remote#329]
When you set `moduleResolution` to `node16` or `nodenext`, you will get 

> Module '"next-mdx-remote"' has no exported member

The problem is 
```ts
// index.d.ts
export * from './dist/index';
```

The fix is 
```ts
// index.d.ts
export * from './dist/index.js';
```

The fix should work even your `moduleResolution` is `node`.
Another way is to fix is specifying them in `exports` of `package.json`.

**Comment by Lord-Valen** [comment]
This is true of `serialize.d.ts` and `rsc.d.ts` also. I'm surprised no one has picked up such a simple issue considering it renders this library unusuable with `node16`/`nodenext` resolution unless you declare your own types (annoying).

**Comment by punkpeye** [comment]
@BRKalow this is a pretty major issue :-(

**Comment by BRKalow** [comment]
Pull Requests are welcome! 🙏

Otherwise, we'll do our best to get to this soon.

**Comment by irontitan76** [comment]
https://github.com/hashicorp/next-mdx-remote/pull/377
